From 98572fc24369058d39012a4aeefa6ec44c567357 Mon Sep 17 00:00:00 2001
From: Ryan Quinn <quinn.ryanx@gmail.com>
Date: Wed, 11 Mar 2026 12:07:42 -0700
Subject: [PATCH] fix onnxruntime

---
 CMakeLists.txt           | 86 +++++++++++++---------------------------
 FastFrames/ONNXWrapper.h |  2 +-
 2 files changed, 29 insertions(+), 59 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9c270d5..b5526fd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -88,7 +88,7 @@ endif()
 
 find_package( Python3 COMPONENTS Development REQUIRED )
 find_package( LCG QUIET )
-find_package( onnxruntime QUIET )
+find_package( onnxruntime CONFIG QUIET )
 
 include_directories(${CMAKE_CURRENT_SOURCE_DIR})
 # ------------------------------------------------
@@ -98,68 +98,34 @@ include_directories(${CMAKE_CURRENT_SOURCE_DIR})
 file(GLOB SOURCES "Root/*.cc" )
 file(GLOB LIB_HEADERS "FastFrames/*.h" )
 
+add_library( FastFrames SHARED ${LIB_HEADERS} ${SOURCES} )
+set_target_properties(FastFrames PROPERTIES SUFFIX ".so")
+target_include_directories(FastFrames
+  PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}>
+)
+target_link_libraries(FastFrames
+  PUBLIC
+    ROOT::Core
+    ROOT::Hist
+    ROOT::RIO
+    ROOT::Tree
+    ROOT::ROOTDataFrame
+    ROOT::ROOTVecOps
+)
+
+set_property(TARGET FastFrames PROPERTY PUBLIC_HEADER ${LIB_HEADERS})
+
 # ------------------------------------------------
 # ONNXWrapper conditional compilation
 # ------------------------------------------------
 
-# Define a variable for the ONNX module path. Allow it to be set from the command line.
-set(ONNXRUNTIME_ROOTDIR "" CACHE PATH "Path to the ONNX Runtime installation directory")
-
-# Check if onnxruntime was found
-if(onnxruntime_FOUND OR NOT ONNXRUNTIME_ROOTDIR STREQUAL "")
-   if (onnxruntime_FOUND)
-      message(STATUS "onnxruntime found at: ${onnxruntime_DIR}")
-   endif()
-   message(STATUS "ONNX Runtime found - Enabling ONNXWrapper")
-   if (NOT ${onnxruntime_FOUND})
-      message(STATUS "ONNXRUNTIME_ROOTDIR: ${ONNXRUNTIME_ROOTDIR}")
-      set(ONNXRUNTIME_INCLUDE_DIRS ${ONNXRUNTIME_ROOTDIR}/include)
-      set(ONNXRUNTIME_LIBRARIES ${ONNXRUNTIME_ROOTDIR}/lib/libonnxruntime.so)
-   endif()
-
-   # This is necessary for Mac installations because
-   # the onnxruntime executable can be found without the paths being set
-   if ((NOT DEFINED ONNXRUNTIME_INCLUDE_DIRS) AND (NOT DEFINED ONNXRUNTIME_LIBRARIES))
-      message(STATUS "ONNXRUNTIME_ROOTDIR not set, need to search for ONNXRUNTIME_INCLUDE_DIRS and ONNXRUNTIME_LIBRARIES")
-      find_path(ONNXRUNTIME_INCLUDE_DIRS
-         NAMES onnxruntime_c_api.h
-         HINTS ${onnxruntime_DIR}/../include
-               /usr/local/include
-               /opt/homebrew/include/onnxruntime
-               /usr/include
-      )
-      message(STATUS "Found ONNXRUNTIME_INCLUDE_DIRS: ${ONNXRUNTIME_INCLUDE_DIRS}")
-
-      find_library(ONNXRUNTIME_LIBRARIES
-         NAMES onnxruntime libonnxruntime
-         HINTS ${onnxruntime_DIR}/../lib
-               /opt/homebrew/include/onnxruntime
-               /opt/homebrew/lib
-               /opt/homebrew/Cellar/
-      )
-      message(STATUS "Found ONNXRUNTIME_LIBRARIES: ${ONNXRUNTIME_LIBRARIES}")
-   endif()
-
-   message(STATUS "ONNXRUNTIME_INCLUDE_DIRS: ${ONNXRUNTIME_INCLUDE_DIRS}")
-   message(STATUS "ONNXRUNTIME_LIBRARIES: ${ONNXRUNTIME_LIBRARIES}")
-   add_definitions(-DONNXRUNTIME_AVAILABLE)
-else()
-   message(STATUS "ONNX Runtime not found - Disabling ONNXWrapper")
-
-   # Remove ONNXWrapper source files from the lists
-   #list(REMOVE_ITEM LIB_HEADERS ${CMAKE_CURRENT_SOURCE_DIR}/FastFrames/ONNXWrapper.h )
-   #list(REMOVE_ITEM SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/Root/ONNXWrapper.cc )
+if(onnxruntime_FOUND)
+  message(STATUS "ONNX Runtime found")
+  target_link_libraries(FastFrames PUBLIC onnxruntime::onnxruntime)
+  target_compile_definitions(FastFrames PUBLIC ONNXRUNTIME_AVAILABLE)
 endif()
 
-# Build the shared library.
-add_library( FastFrames SHARED ${LIB_HEADERS} ${SOURCES} )
-set_target_properties(FastFrames PROPERTIES SUFFIX ".so")
-target_include_directories( FastFrames PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}> $<INSTALL_INTERFACE:include> ${ONNXRUNTIME_INCLUDE_DIRS} )
-target_include_directories(FastFrames SYSTEM PUBLIC ${ONNXRUNTIME_INCLUDE_DIRS} )
-target_link_libraries( FastFrames PUBLIC ROOT::Core ROOT::Hist ROOT::RIO ROOT::Tree ROOT::ROOTDataFrame ROOT::ROOTVecOps ${ONNXRUNTIME_LIBRARIES} )
-set_property( TARGET FastFrames
-   PROPERTY PUBLIC_HEADER ${LIB_HEADERS} )
-
 # Build the Python module for C++ logger
 add_library( cppLogger SHARED python_wrapper/utils/Logger.cxx )
 set_target_properties(cppLogger PROPERTIES SUFFIX ".so")
@@ -180,7 +146,11 @@ set_target_properties( ConfigReaderCpp PROPERTIES
 set(SETUP ${CMAKE_CURRENT_BINARY_DIR}/setup.sh)
 file(WRITE ${SETUP} "#!/bin/bash\n")
 file(APPEND ${SETUP} "# this is an auto-generated setup script\n" )
-file(APPEND ${SETUP} "export LD_LIBRARY_PATH=\${LD_LIBRARY_PATH}:${CMAKE_INSTALL_PREFIX}/lib\n")
+if(APPLE)
+  file(APPEND ${SETUP} "export DYLD_LIBRARY_PATH=\${DYLD_LIBRARY_PATH}:${CMAKE_INSTALL_PREFIX}/lib\n")
+else()
+  file(APPEND ${SETUP} "export LD_LIBRARY_PATH=\${LD_LIBRARY_PATH}:${CMAKE_INSTALL_PREFIX}/lib\n")
+endif()
 file(APPEND ${SETUP} "export PATH=\${PATH}:${CMAKE_CURRENT_SOURCE_DIR}/python\n")
 file(APPEND ${SETUP} "export FASTFRAMES_DIR=${CMAKE_CURRENT_SOURCE_DIR}\n")
 
diff --git a/FastFrames/ONNXWrapper.h b/FastFrames/ONNXWrapper.h
index c88ab47..66ad648 100644
--- a/FastFrames/ONNXWrapper.h
+++ b/FastFrames/ONNXWrapper.h
@@ -10,7 +10,7 @@
 
 #include "FastFrames/Logger.h"
 
-#include <onnxruntime_cxx_api.h>
+#include <onnxruntime/core/session/onnxruntime_cxx_api.h>
 #include <vector>
 
 /**
-- 
2.53.0

