diff --git a/python/check_duplicate_events.py b/python/check_duplicate_events.py
old mode 100644
new mode 100755
index e221e6d..e78a8ec
--- a/python/check_duplicate_events.py
+++ b/python/check_duplicate_events.py
@@ -1,4 +1,4 @@
-#!/bin/env python3
+#!/usr/bin/env python3
 
 """!Script for checking duplicate events in the input root files, based on runNumber and eventNumber.
 """
diff --git a/python/copyDataToScratch.py b/python/copyDataToScratch.py
old mode 100644
new mode 100755
index 5c4dbbe..e0e041f
--- a/python/copyDataToScratch.py
+++ b/python/copyDataToScratch.py
@@ -1,4 +1,6 @@
-# Import necesary modules
+#!/usr/bin/env python3
+
+# Import necessary modules
 import os
 import sys
 import yaml
diff --git a/python/merge_empty_grid_files.py b/python/merge_empty_grid_files.py
old mode 100644
new mode 100755
index 6e2c20a..d126a24
--- a/python/merge_empty_grid_files.py
+++ b/python/merge_empty_grid_files.py
@@ -1,4 +1,4 @@
-#!/bin/env python3
+#!/usr/bin/env python3
 
 """!Script for merging files from GRID with empty trees - this is needed to avoid the problem with empty trees in the GRID files - RDF cannot properly handle them
 """
diff --git a/python/merge_region_jobs.py b/python/merge_region_jobs.py
old mode 100644
new mode 100755
diff --git a/python/produce_filelist.py b/python/produce_filelist.py
index a16a802..fbfc5f2 100755
--- a/python/produce_filelist.py
+++ b/python/produce_filelist.py
@@ -1,4 +1,4 @@
-#!/bin/env python3
+#!/usr/bin/env python3
 
 """!Script for producing filelist.txt from a folder containing root files, it can be also imported as a module.
 """
diff --git a/python/produce_metadata_files.py b/python/produce_metadata_files.py
index aad1899..47f9de0 100755
--- a/python/produce_metadata_files.py
+++ b/python/produce_metadata_files.py
@@ -1,4 +1,4 @@
-#!/bin/env python3
+#!/usr/bin/env python3
 
 """!Script to produce filelist.txt and sum_of_weights.txt files from a folder containing root files.
 It will loop over all root files in root_files_folder, and produce filelist and sum_of_weights files for all samples, where sample is characterized by (dsid, campaign, data_type).
diff --git a/python/produce_sum_weights_file.py b/python/produce_sum_weights_file.py
index 056f13f..679fe59 100755
--- a/python/produce_sum_weights_file.py
+++ b/python/produce_sum_weights_file.py
@@ -1,4 +1,4 @@
-#!/bin/env python3
+#!/usr/bin/env python3
 
 """
 Script/module to produce a sum of weights file from a filelist.
diff --git a/python/produce_trexfitter_config.py b/python/produce_trexfitter_config.py
old mode 100644
new mode 100755
index 2019486..c6ffd12
--- a/python/produce_trexfitter_config.py
+++ b/python/produce_trexfitter_config.py
@@ -1,3 +1,5 @@
+#!/usr/bin/env python3
+
 """
 @file Script for producing TRExFitter config file
 """
diff --git a/python/skim_filelist.py b/python/skim_filelist.py
old mode 100644
new mode 100755
index 5b72a92..c6b92f0
--- a/python/skim_filelist.py
+++ b/python/skim_filelist.py
@@ -1,8 +1,8 @@
+#!/usr/bin/env python3
 
 """!Script for skimming filelists by removing files with no reco-level events. It cannot be used before producing sum of weights files, nor in the case when truth trees are needed.
 """
 
-
 from sys import argv
 from ROOT import TFile
 
